### PR TITLE
Bluetooth: shell: Transition the remaining parts to the new shell

### PR DESCRIPTION
**Pull request by carlescufi**

### CODE REVIEW

**Comment by carlescufi** [comment]
cc @rakons

**Comment by carlescufi** [comment]
Build will fail until #10322 is merged.

File: subsys/bluetooth/shell/flash.c [hunk 1]
```diff
@@ -73,17 +79,17 @@ static int cmd_flash(int argc, char *argv[])
 
 	flash_dev = device_get_binding(FLASH_DEV_NAME);
 	if (!flash_dev) {
-		printk("Nordic nRF5 flash driver was not found!\n");
+		error(shell, "Nordic nRF5 flash driver was not found!\n");
```
Comment (jarz-nordic) [inline subsys/bluetooth/shell/flash.c]: I think `error` macro already adds `\r\n` at the end of a message.

So you will have one extra empty line.

File: subsys/bluetooth/shell/flash.c [hunk 2]
```diff
@@ -55,15 +61,15 @@ static int cmd_erase(int argc, char *argv[])
 	result = flash_erase(flash_dev, page_addr, size);
 
 	if (result) {
-		printk("Erase Failed, code %d.\n", result);
+		error(shell, "Erase Failed, code %d.\n", result);
 	} else {
-		printk("Erase success.\n");
+		print(shell, "Erase success.\n");
```
Comment (jarz-nordic) [inline subsys/bluetooth/shell/flash.c]: I think `print` macro already adds `\r\n` at the end of a message.

So you will have one extra empty line.

File: subsys/bluetooth/shell/ll.c [hunk 3]
```diff
@@ -218,22 +222,22 @@ int cmd_scanx(int argc, char *argv[])
...
 	err = ll_scan_enable(enable);
 	if (err) {
 		goto exit;
 	}
 
 exit:
-	printk("done (err= %d).\n", err);
+	print(shell, "done (err= %d).\n", err);
 
 	return 0;
```
Comment (jarz-nordic) [inline subsys/bluetooth/shell/ll.c]: This implementation can return 0 on `err` condition.

**Review by jakub-uC — CHANGES_REQUESTED** [review]

**Comment by jarz-nordic** [comment]
@carlescufi : general remark.
Macros `print` and `error` already adds `\r\n`

**Comment by carlescufi** [comment]
@jarz-nordic comments addressed, rebased against latest master

File: subsys/bluetooth/shell/flash.c [hunk 4]
```diff
@@ -146,16 +152,16 @@ static int cmd_read(int argc, char *argv[])
 		u32_t data;
 
 		flash_read(flash_dev, addr, &data, sizeof(data));
-		printk("0x%08x ", data);
+		print(shell, "0x%08x ", data);
 		addr += sizeof(data);
 	}
 
-	printk("\n");
+	print(shell, "");
```
Comment (jarz-nordic) [inline subsys/bluetooth/shell/flash.c]: looks like obsolete piece of code

File: subsys/bluetooth/shell/flash.c [hunk 5]
```diff
@@ -192,31 +198,49 @@ static int cmd_test(int argc, char *argv[])
...
-static const struct shell_cmd flash_commands[] = {
-	{ "flash-write", cmd_flash, "<address> <Dword> <Dword>..."},
-	{ "flash-erase", cmd_erase, "<page address> <size>"},
-	{ "flash-read", cmd_read, "<address> <Dword count>"},
-	{ "flash-test", cmd_test, "<address> <size> <repeat count>"},
-	{ NULL, NULL, NULL}
+#define HELP_NONE "[none]"
+
+SHELL_CREATE_STATIC_SUBCMD_SET(flash_cmds) {
+	SHELL_CMD(write, NULL, "<address> <dword> <dword>...", cmd_write),
```
Comment (jarz-nordic) [inline subsys/bluetooth/shell/flash.c]: These subcommands must be added in alphabetical order. Otherwise tab key may not work correctly.

**Review by jakub-uC — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/shell/flash.c (hunk 4 above)
Comment (carlescufi) [reply subsys/bluetooth/shell/flash.c]: No, it's there to print an empty line

File: subsys/bluetooth/shell/flash.c (hunk 5 above)
Comment (carlescufi) [reply subsys/bluetooth/shell/flash.c]: Fixed, thank you!

**Review by jakub-uC — APPROVED** [review]

**Pull request merged**